R/html.R

```diff
@@ -665,7 +665,7 @@ parse_fig_labels = function(content, global = FALSE) {
       }
       labs[[i]] = label_prefix(type, sep = ': ')(num)
       k = max(figs[figs <= i])
-      content[k] = paste(c(content[k], sprintf('<span id="%s"></span>', lab)), collapse = '')
+      content[k] = paste(c(content[k], sprintf('<span style="display:block;" id="%s"></span>', lab)), collapse = '')
```

Comment:
I'm wondering if it's okay to use `div` instead:

```suggestion
      content[k] = paste(c(content[k], sprintf('<div id="%s"></div>', lab)), collapse = '')
```

Reply:
I have seen some example using a span for other issues(e.g https://stackoverflow.com/a/29123823/3436535)
I don't know if using a div can break something as it is stronger than span for HTML structure, isn't it ? 

Also, changing it would break some custom CSS if any user has used a rule based on the span, but it seems unlikely (I don't see why it would be useful)

Question: idealy, adding the `id` on the `img` tag would be the best, wouldn't it ?  or on the `div` of class figure ? 

Anyway, I think a `div` could work yes. I did not thought of that. 

maybe with `visibility = "hidden"` ? or anything that could help for Accessibility - I wonder how this empty tag renders.

Reply:
> So it is safer for now to add a special div or span for our id to avoid conflict right ?

That's what I think.

> We may think more about this if we reimplement the referencing feature based on Lua filter or else.

My impression is that they already add the id to the figure environment instead of using an invisible anchor span, but I could be wrong (you can double check).

> So let's go with a div ? I am fine with it.

Yes, we can revert back to a block span later if we discover any problems.

Reply:
> My impression is that they already add the id to the figure environment instead of using an invisible anchor span, but I could be wrong (you can double check).

They put the id to the img tag if you add an id using link attributes
````powershell
❯ pandoc -t html4
![cap](https://rmarkdown.rstudio.com/docs/reference/figures/logo.png){#id}
^Z
<div class="figure">
<img src="https://rmarkdown.rstudio.com/docs/reference/figures/logo.png" id="id" alt="" />
<p class="caption">cap</p>
</div>
````

Reply:
Okay I see. Thanks!